rclcpp_action/test/test_server.cpp

```diff
@@ -95,6 +95,32 @@ TEST_F(TestServer, construction_and_destruction)
   (void)as;
 }
 
+TEST_F(TestServer, construction_and_destruction_callback_group)
```

Comment:
There is no `EXPECT_*` or `ASSERT_*` in this test. What are you trying to test?

Reply:
Like the construction_and_destruction test above, it just simply checks that creating and cleaning up the object doesn't throw exceptions or cause segfaults. I didn't really have any other ideas for how to add coverage for the callback group constructor other than this simple test.

I'm open to ideas for better exercising that constructor though.

Reply:
you can do something like:

```
  try{
     <test code>
  }
  catch (Exception& e)
  {
    FAIL();
  }

  SUCCEED();

Reply:
Added an ASSERT_NO_THROW around this code, which does the same thing.